subprojects/build-cache-packaging/src/main/java/org/gradle/caching/internal/packaging/impl/TarBuildCacheEntryPacker.java

```diff
@@ -203,7 +201,7 @@ public String apply(@Nullable CacheableTree tree) {
 
                 boolean missing = matcher.group(1) != null;
                 String childPath = matcher.group(3);
-                tarEntry = unpackTree(tree, tarInput, tarEntry, childPath, missing, snapshots, entries);
+                tarEntry = unpackTree(treeName, tree.getType(), tree.getRoot(), tarInput, tarEntry, childPath, missing, snapshots, entries);
```

Comment:
Should we use a `MutableLong` for entries here instead of an `AtomicInteger`?

Reply:
Good point. Why did we use an integer in any case?